core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -92,6 +92,11 @@ object SparkSubmit extends CommandLineUtils with Logging {
 
   private val CLASS_NOT_FOUND_EXIT_STATUS = 101
 
+  // Following constants are visible for testing.
+  private[deploy] val YARN_SUBMIT_CLASS = "org.apache.spark.deploy.yarn.YarnClusterApplication"
+  private[deploy] val REST_SUBMIT_CLASS = classOf[RestSubmissionClientApp].getName()
+  private[deploy] val STANDALONE_SUBMIT_CLASS = classOf[ClientApp].getName()
```

Comment:
Maybe we can rename these variables to `XXX_CLUSTER_SUBMIT_CLASS`.

Reply:
Sure.